### PR TITLE
Atomic Compare Test

### PR DESCRIPTION
**Pull request by nolanbaker31**
Does not result in errors, but I'm unsure if implementation properly tests atomic compare, as without it would still most likely not give errors.

### CODE REVIEW

**Comment by tmh97** [comment]
Tentative: May need to give credit to OpenMP examples doc

File: tests/5.1/atomic/test_atomic_compare.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+   for(int i=0; i<N; i++){
+      arr[i] = rand()%1000;
+   }
+   for(int i = 0; i<N; i++){ // Sets max through non-parallel methods
+      if(arr[i] > max){
+         max = arr[i];
+      }
+   }
+   #pragma omp parallel for // Sets max using parallel for loop, using atomic to ensure max is correct
```
Comment (tmh97) [inline tests/5.1/atomic/test_atomic_compare.c:32]: pmax should be a shared variable

Comment (tmh97) [reply tests/5.1/atomic/test_atomic_compare.c:32]: After making this change `#pragma omp paralle for shared(pmax)` and testing with cray, clang, and gcc this should be good to go.

**Comment by tmh97** [comment]
@nolanbaker31 changes look good, were you able to test this on a few different compilers?

**Review by spophale — APPROVED** [review]
I think it does stress the atomic else the max value would be overwritten and the test would give incorrect answers. Such tests need to be run more than once to catch race (if atomic is not correctly implemented).
Also need a device version.

**Pull request merged**